File: Arrays-Hashing/twoSum/Go/twoSum.go
package main

func twoSum(nums []int, target int) []int {
	numMap := make(map[int]int)

	for i, num := range nums {
		complement := target - num
		if j, ok := numMap[complement]; ok {
			return []int{j, i}
		}
		numMap[num] = i
	}

	return []int{}
}

File: twoSum/Php/twoSum.php
<?php

class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum(array $nums, int $target): array {
        $hashMap = [];
        foreach ($nums as $key => $num) {
            $difference = $target - $num;
            if(isset($hashMap[$difference])) {
                return [$key, $hashMap[$difference]];
            }
            $hashMap[$num] = $key;
        }
        return [];
    }
}